handlers/people-handler/people_handler.py

```diff
@@ -0,0 +1,484 @@
+            for word in sentence_split:
+                if (word == verb or passed):
+                    passed = True
+                    sentence += word
+                    sentence += " "
+                else:
+                    continue
+            sentence = re.sub('[^A-Za-z0-9]+', ' ', sentence)
+            rendering += sentence
+    else:
+        rendering, emo_count, clothes_count = m.rendering_for_multiple_people(
+            object_emotion_inanimate, rendering, preprocessors)
+        sentence = re.sub('[^A-Za-z0-9]+', ' ', res)
+        if (emo_count > 0 and clothes_count > 0):
+            rendering += ". Interpretation: " + sentence
+        elif (emo_count == 0 and clothes_count == 0):
+            rendering = "Image possibly contains: " + sentence
+        else:
+            nlp = spacy.load("en_core_web_sm")
+            doc = nlp(res)
+            verb_posi = [token.i for token in doc if token.pos_ == "VERB"]
+            posi = verb_posi[-1]
+            sentence = ""
+            i = 0
+            for token in doc:
+                if (i >= posi):
+                    sentence += token.orth_
+                    sentence += " "
+                i += 1
+            sentence = re.sub('[^A-Za-z0-9]+', ' ', sentence)
+            rendering += sentence
+
+    print("final output is: ")
+    print(rendering)
+    print("\n")
+    return rendering
+
+
+@app.route("/handler", methods=["POST"])
+def handle():
+    logging.debug("Received request")
+    contents = request.get_json()
+
+    # Check preprocessor data
+    preprocessors = contents['preprocessors']
+    possible_people = 0
+
+    # No Object Detector found
+    prep = "ca.mcgill.a11y.image.preprocessor.objectDetection"
+    if prep not in preprocessors:
+        logging.debug("No Object Detector found")
+        logging.debug("Sending response")
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        return response
+
+    if ("ca.mcgill.a11y.image.preprocessor.graphicTagger" in preprocessors):
+        if (preprocessors["ca.mcgill.a11y.image.preprocessor.graphicTagger"]
+                ["category"] == "people"):
+            objects = preprocessors[prep]["objects"]
+            possible_people += 1
+
+    # checks if emotion detection detects face
+    possible_people += c.custom_check(preprocessors)
+    objects = preprocessors[prep]["objects"]
+    res = preprocessors["ca.mcgill.a11y.image.preprocessor.caption"]["caption"]
+    if (possible_people <= 1):
+        response = {
+            "request_uuid": contents["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": [
+                {
+                    "type_id": "ca.mcgill.a11y.image.renderer.Text",
+                    "description": "The text found in a graphic.",
+                    "data": {
+                        "text": "Cannot be rendered"
```

Comment:
You don't return renderings for error cases!

Reply:
@jeffbl what should be returned in this case then?

Reply:
See example: https://github.com/Shared-Reality-Lab/IMAGE-server/blob/f3ead6b9f21a7785edd86572494e56146d8205ed/handlers/svg-action-recognition/ar_svg.py#L151